core/templates/pages/android-page/android-page.component.ts

```diff
@@ -135,6 +143,22 @@ export class AndroidPageComponent implements OnInit, OnDestroy {
     this.featuresShown = featureNumber;
   }
 
+  validateEmailAddress(): boolean {
+    // A simple check for basic email validation.
+    let regex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
+    return regex.test(this.emailAddress);
+  }
+
+  subscribeToAndroidList(): void {
+    this.androidUpdatesBackendApiService.subscribeUserToAndroidList(
+      this.emailAddress, this.name
+    ).then(() => {
+      this.alertsService.addInfoMessage('Done!', 1000);
+    }).catch(errorResponse => {
+      // This is placed here in order to satisfy Unit tests.
```

Comment:
This doesn't seem like a particularly good reason IMO :P

Why do the unit tests require it? I think all async requests should have a catch handler in general -- but what do you actually plan to do if there's an error? Shouldn't we show the user something? A video recording of the experience here would be great.

Reply:
Oh, I took this from the learner-dashboard page where this comment and catch section was present.
The only backend error here would be the various errors that may occur mailchimp's side, which would have to be aggregated into some message "Signup wasn't possible, please try again" for the user which isn't very useful as trying again probably won't work either in that case, so I didn't add any error message handling. Should I add just a "request errored" type of message anyway?

Reply:
Done, added info messages for the error cases.